var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -127,7 +134,9 @@ def install(self, spec, prefix):
                 'complex' if '+complex' in spec else 'real'),
             '--with-shared-libraries=%s' % ('1' if '+shared' in spec else '0'),
             '--with-debugging=%s' % ('1' if '+debug' in spec else '0'),
-            '--with-blas-lapack-dir=%s' % spec['lapack'].prefix
+            '--with-lapack-lib=%s' % spec['lapack'].lapack_static_lib,
```

Comment:
Apparently, I switched to static libs to get PISM building.  See here, it is unfortunately not very explanatory:
   https://github.com/citibeth/spack/commit/f7700e23ebedd3f01e4005ce30f3a063b3713c3d

> Better yet, change to lapack_shared_lib and blas_shared_lib if building shared petsc, otherwise use static

I submitted a PR last night for another package that does this.  Not sure why this package did not.  Unless we decide to turn those exported libs into tuples/lists, I think this is the right approach.